### ISSUE TITLE
[Notification] We shouldn't retry sending slack if it has error "channel_not_found"

### ISSUE THREAD

**Issue by qnkhuat** [metabase/metabase#49647]
### Describe the bug

When sending notifications to a slack channel that's no longer exists, we keep retrying until the retry attempts are maxed-out.
The retry mechanism includes uploading the assets and posting the message.

This is necessary, we should stop sending the channel at the first attempt and have the task_history.status = failed.

### To Reproduce

1. On slack: create a channel "testing-alert" 
2. On metabase: Setup slack connection
3. Create an alert that send to "testing-alert" channel
4. Delete the "testing-alert" channel
5. Trigger sending the "alert"

### Information about your Metabase installation

master

### Severity

p3

**Comment by harrylewis** [comment]
Hey @qnkhuat 👋🏻 

I'm new to the codebase and was trying to figure out where in the code this would require changes. Would it be in this file?

https://github.com/metabase/metabase/blob/cd209344c6175fbd8bb9b4b5a9ac5565d96f420e/src/metabase/notification/send.clj#L53-L56

**Comment by qnkhuat** [comment]
hey @harrylewis, `metabase.notification.send` is where we decide what to do with the error from each channel. you also need to look at https://github.com/metabase/metabase/blob/e2fd234cc27e568016bdff03561fa328664040d3/src/metabase/integrations/slack.clj#L118 , this is where we classify errors from slack.

**Comment by amuradyan** [comment]
Hi guys. If this is still valid, /and from the code it looks like it is/ I'd like to pick it up.

@qnkhuat do you expect `handle-error` to produce something like in case of an invalid token like below, and update `should-retry-sending?` accordingly? If so, do we need a `slack-channel-error-codes` or should we handle only the `channel_not_found`?

```clojure
 {:error-code (:error body)
  :errors     {:slack-channel message}}
```

P.S.
I couldn't find a slack or discord or zulip or anything to chat with you metabase guys. Can you point me to one, if such exists?

**Comment by paoliniluis** [comment]
@amuradyan we don't have a public channel, only through github. If you want to pick it up please remember to add enough tests to the PR

**Comment by qnkhuat** [comment]
@amuradyan yes, we need to classify the error in `handle-error` and the skip it in `should-retry-sending?`.

**Comment by amuradyan** [comment]
@qnkhuat @paoliniluis  Well guys, the PR is open. I am sorry for force pushes, I hope that won't be an issue.

I had some hard time figuring out how to run just the tests that have failed and in a watch mode, so if you have the answer, please share.
